VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room.component.ts

```diff
@@ -0,0 +1,689 @@
+            lock: lock
+        });
+        await this.consultationService.lockConsultation(this.vhConference.id, roomLabel, lock);
+    }
+
+    closeStartPrivateConsultationModal() {
+        this.displayStartPrivateConsultationModal = false;
+    }
+
+    closeJoinPrivateConsultationModal() {
+        this.displayJoinPrivateConsultationModal = false;
+    }
+
+    toggleAccordian() {
+        this.privateConsultationAccordianExpanded = !this.privateConsultationAccordianExpanded;
+    }
+
+    setTrapFocus() {
+        ModalTrapFocus.trap('video-container');
+    }
+
+    dismissJoinHearingWarning() {
+        this.showJoinHearingWarning = false;
+        this.setUpSubscribers();
+    }
+
+    onLeaveHearingButtonClicked() {
+        this.displayLeaveHearingPopup = true;
+    }
+
+    leave(confirmation: boolean) {
+        this.displayLeaveHearingPopup = false;
+        if (!confirmation) {
+            return;
+        }
+        const feedbackUrl = 'https://www.smartsurvey.co.uk/s/VideoHearings_Feedback/';
+        window.location.assign(feedbackUrl);
+    }
+
+    init() {
+        this.divTrapId = 'video-container';
+
+        if (this.userRole === WaitingRoomUserRole.Judge) {
+            this.initForJudge();
+            return;
+        }
+
+        this.destroyedSubject = new Subject();
+
+        this.unloadDetectorService.shouldUnload.pipe(takeUntil(this.destroyedSubject)).subscribe(() => this.onShouldUnload());
+        this.unloadDetectorService.shouldReload.pipe(take(1)).subscribe(() => this.onShouldReload());
+
+        this.errorCount = 0;
+        this.logger.debug(`${this.componentLoggerPrefix} loading waiting room`);
+        this.connected = false;
+        this.getConference();
+        if (this.deviceTypeService.isHandheldIOSDevice()) {
+            this.showJoinHearingWarning = true;
+        } else {
+            this.setUpSubscribers();
+        }
+    }
+
+    initForJudge() {
+        if (!this.vhConference || !this.vhParticipant) {
+            return;
+        }
+
+        this.errorCount = 0;
+        this.logger.debug(`${this.componentLoggerPrefix} Loading judge waiting room`);
+        this.loggedInUser = this.route.snapshot.data['loggedUser'];
+
+        this.unloadDetectorService.shouldUnload.pipe(takeUntil(this.onDestroy$)).subscribe(() => this.onShouldUnload());
+        this.unloadDetectorService.shouldReload.pipe(take(1)).subscribe(() => this.onShouldReload());
+
+        try {
+            this.logger.debug(`${this.componentLoggerPrefix} Defined default devices in cache`);
+            this.connected = false;
+            this.getConference();
+            this.subscribeToClock();
+            this.startEventHubSubscribers();
+            this.connectToPexip();
+        } catch (error) {
+            this.logger.error(`${this.componentLoggerPrefix} Failed to initialise the judge waiting room`, error);
+            const conferenceId = this.route.snapshot.paramMap.get('conferenceId');
```

Comment:
you don't need this line anymore, since you have VHConference you already have the ID from so you don't need to pull it from the route anymore

Reply:
updated, thanks